exec/cnex/cell.c

```diff
@@ -342,11 +401,14 @@ int32_t cell_compareCell(const Cell * s, const Cell * d)
     }
 
     switch (s->type) {
-        case cString:   return string_compareString(s->string, d->string);
-        case cAddress:  return s->address == d->address != 0;
-        case cBoolean:  return s->boolean == d->boolean != 0;
-        case cArray:    return s->array_size == d->array_size != 0;
-        case cNumber:   return bid128_quiet_equal(s->number, d->number) != 0;
+        case cAddress:      return s->address != d->address;
+        case cArray:        return s->array_size != d->array_size;
+        case cBoolean:      return s->boolean != d->boolean;
+        case cDictionary:   return !dictionary_compareDictionary(s->dictionary, d->dictionary);
+        case cNothing:      return s != d;
+        case cNumber:       return !number_is_equal(s->number, d->number);
+        case cPointer:      return (void*)s->address != (void*)d->address;
```

Comment:
Oh.  I'm not sure that this is exactly what I expected out of this either.  However, without the casts, it would still do what I want it to do.  So, good catch.